pkg/server/config.go

```diff
@@ -287,33 +287,33 @@ func setOpenFileLimit(physicalStoreCount int) (int, error) {
 	}
 
 	// The max open file descriptor limit is too low.
-	if rLimit.Max < minimumOpenFileLimit {
+	if uint64(rLimit.Max) < minimumOpenFileLimit {
 		return 0, fmt.Errorf("hard open file descriptor limit of %d is under the minimum required %d\n%s",
-			rLimit.Max,
+			uint64(rLimit.Max),
 			minimumOpenFileLimit,
 			productionSettingsWebpage)
 	}
 
 	// If current open file descriptor limit is higher than the recommended
 	// value, we can just use the default value.
-	if rLimit.Cur > recommendedOpenFileLimit {
+	if uint64(rLimit.Cur) > recommendedOpenFileLimit {
 		return engine.DefaultMaxOpenFiles, nil
 	}
 
 	// If the current limit is less than the recommended limit, set the current
 	// limit to the minimum of the max limit or the recommendedOpenFileLimit.
 	var newCurrent uint64
-	if rLimit.Max > recommendedOpenFileLimit {
+	if uint64(rLimit.Max) > recommendedOpenFileLimit {
 		newCurrent = recommendedOpenFileLimit
 	} else {
-		newCurrent = rLimit.Max
+		newCurrent = uint64(rLimit.Max)
 	}
-	if rLimit.Cur < newCurrent {
+	if uint64(rLimit.Cur) < newCurrent {
 		if log.V(1) {
 			log.Infof(context.TODO(), "setting the soft limit for open file descriptors from %d to %d",
-				rLimit.Cur, newCurrent)
+				uint64(rLimit.Cur), newCurrent)
 		}
-		rLimit.Cur = newCurrent
+		rLimit.Cur = int64(newCurrent)
```

Comment:
This breaks the non-FreeBSD builds. Rather than the approach you took here, I would create `rlimit_unix.go` and `rlinux_freebsd.go` files and define small wrappers around `syscall.{Get,Set}rlimit` that have a consistent interface. I can provide more details if that explanation is insufficient.